### PR TITLE
feat(fcm): Add `sendEach` and `sendEachForMulticast` for FCM batch send

### PR DESCRIPTION
**Pull request by Doris-Ge**
- Deprecate `sendAll()` and `sendMulticast()` APIs.
- Add new `sendEach()` and `sendEachForMulticast()` APIs.

### CODE REVIEW

File: src/messaging/messaging.ts [hunk 1]
```diff
@@ -250,6 +251,124 @@ export class Messaging {
...
 
+  /**
+  * Sends each message in the given array via Firebase Cloud Messaging.
+  *
+  * Unlike {@link Messaging.sendAll}, this method makes a single RPC call for each message
+  * in the given array.
+  *
+  * The responses list obtained from the return value corresponds to the order of `messages`.
+  * An error from this method or a `BatchResponse` with all failures indicates a total failure
+  * -- i.e. none of the messages in the list could be sent. Partial failures or no failures
```
Comment (egilmorez) [inline src/messaging/messaging.ts:262]: We avoid i. e., so for this I'd suggest just "...indicates a total failure, mean that none of the messages in the list could be sent.

File: src/messaging/messaging.ts [hunk 2]
```diff
@@ -250,6 +251,124 @@ export class Messaging {
...
+
+  /**
+   * Sends the given multicast message to all the FCM registration tokens
+   * specified in it.
+   *
+   * This method uses the {@link Messaging.sendEach} API under the hood to send the given
+   * message to all the target recipients. The responses list obtained from the
+   * return value corresponds to the order of tokens in the `MulticastMessage`.
+   * An error from this method or a `BatchResponse` with all failures indicates a total failure
+   * -- i.e. none of the messages in the list could be sent. Partial failures or no failures
```
Comment (egilmorez) [inline src/messaging/messaging.ts:332]: Same suggestion as above RE "-- i.e."

**Review by egilmorez — APPROVED** [review]
LG with one optional nit, thanks!

**Comment by Doris-Ge** [comment]
Thanks Eric for reviewing the PR! @lahirumaramba I tried to push the fix to the fcm-batch-send branch but it got rejected because "Protected branch update failed for refs/heads/fcm-batch-send". Should I create another PR from dorisge/fcm-batch-send for the fix and merge it to fcm-batch-send?

File: src/messaging/messaging.ts (hunk 2 above)
Comment (egilmorez) [reply src/messaging/messaging.ts:332]: Oh I'm kind of glad you got that failure, as I see I have a typo in my comment!!

It should read:

"We avoid i. e., so for this I'd suggest just "...indicates a total failure, meaning that none of the messages in the list could be sent."

Note "meaning" with "ing," thanks!

Comment (Doris-Ge) [reply src/messaging/messaging.ts:332]: Thank you for noticing that! I also had the question and should've asked.  I'll fix that.

**Review by lahirumaramba — APPROVED** [review]
Thank you! LGTM

**Pull request merged**